auth/src/main/java/com/firebase/ui/auth/ui/HelperActivityBase.java

```diff
@@ -42,6 +51,40 @@ protected void onCreate(Bundle savedInstance) {
         super.onCreate(savedInstance);
         mAuthHelper = new AuthHelper(getFlowParams());
         mProgressDialogHolder = new ProgressDialogHolder(this);
+
+        getFlowHolder().getIntentStarter().observe(this, new Observer<Pair<Intent, Integer>>() {
+            @Override
+            public void onChanged(@Nullable Pair<Intent, Integer> request) {
+                if (request == null) {
+                    throw new IllegalStateException("Cannot start null request");
+                }
+
+                startActivityForResult(request.first, request.second);
+            }
+        });
+        getFlowHolder().getPendingIntentStarter()
```

Comment:
How do we make sure we only observe the same `Pair` once?  Apologies, I am sort of an architecture components noob.

Reply:
That's what the `SingleLiveEvent` is for: it keeps track of which observers have received the update and then doesn't send it again if the observer already got it. In addition, lifecycle owners (activities) that aren't past `onStart` don't receive updates anyway.

No problemo about the questions, I'm happy to share what I've learned. 😄